src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Write.HandleEnumerable.cs

```diff
@@ -48,16 +48,36 @@ public static partial class JsonSerializer
 
             if (state.Current.Enumerator != null && state.Current.Enumerator.MoveNext())
             {
+                // If the enumerator contains typeof(object), get the run-time type
+                if (elementClassInfo.ClassType == ClassType.Object && jsonPropertyInfo.ElementClassInfo.Type == typeof(object))
+                {
+                    object currentValue = state.Current.Enumerator.Current;
+                    if (currentValue != null)
+                    {
+                        Type runtimeType = currentValue.GetType();
+                        
+                        // Ignore object() instances since they are handled as an empty object.
+                        if (runtimeType != typeof(object))
```

Comment:
Is this saying that `new[] { new object() }` makes the empty array?  That seems weird to me.

Reply:
This is for an element, so it's saying if the element instance is `typeof(object)` don't do any extra work since the code below is going to treat the type as `typeof(object)` anyway. The `if` isn't needed - it just prevents a special nonperformance case.

A `new[] { new object() }` works as expected and returns `[{}]`